state/firewallrules.go

```diff
@@ -90,20 +90,21 @@ func NewFirewallRules(st *State) *firewallRulesState {
 }
 
 // Save stores the specified firewall rules.
-func (fw *firewallRulesState) Save(service WellKnownServiceType, whiteListCidrs, blackListCidrs []string) (*FirewallRule, error) {
-	if err := service.validate(); err != nil {
-		return nil, errors.Trace(err)
+func (fw *firewallRulesState) Save(rule *FirewallRule) error {
```

Comment:
please make it a non-pointer type. then you can be sure it's non-nil, and the caller can be sure that this method won't (can't) modify it

Reply:
ok